operator/api/v1beta1/helmrelease_types.go

```diff
@@ -26,6 +26,20 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type HelmReleaseStatusReason string
+
+const (
+	ReasonDeleteError     HelmReleaseStatusReason = "DeleteError"
```

Comment:
I think these status reasons are necessary to be able to handle cycles in the FSM graph of the controller's logic.

For example, we could reach a `FAILED` deployment state by either a failed install (can't rollback), or a failed upgrade (should rollback), or a failed rollback (would be handled specially, since a rollback to a known good version shouldn't ever fail).